pennylane/compiler/qjit_api.py

```diff
@@ -411,11 +411,11 @@ def _get_while_loop_qfunc_prim():
     while_loop_prim.multiple_results = True
 
     @while_loop_prim.def_impl
-    def _(*jaxpr_args, jaxpr_body_fn, jaxpr_cond_fn, n_consts_body, n_consts_cond):
+    def _(*args, jaxpr_body_fn, jaxpr_cond_fn, body_consts, cond_consts, args_slice):
```

Comment:
`body_consts` (and `cond_consts`) reads like it will be an iterable containing constants for the body, not a slice.
This is underlined by the fact that `args_slice` has the "slice" in it explicitly. I think it would be good to have similar names for the three if they all are slices into `args`.
Just spitballing a few possibilities here that come to mind:
- (`body_consts_sl`, `cond_consts_sl`, `args_sl`)
- (`body_slice`, `cond_slice`, `args_slice`) (would be my favourite of these)
- (`slice_body_consts`, `slice_cond_consts`, `slice_args`)

Reply:
renaming to `body_slice`, `cond_slice`, and `args_slice`.